### PR TITLE
BFD-3175: Encode bfd-mgmt-cmk in Terraform

### PR DESCRIPTION
**Pull request by aadedejifearless**
**JIRA Ticket:**
[BFD-3175](https://jira.cms.gov/browse/BFD-3175)

**User Story or Bug Summary:**

---

On 1/10/24's deployment of BFD 2.93.0, we came across issues with EC2 instances failing to startup. In Cloudwatch we found that the bfd-mgmt-ec2-failing-instances alarm was failing to send to the SNS topic due to the inability to decrypt the KMS key. See attachment screenshot. The policy attached to this KMS key has been altered via ClickOps to allow the decryption of the current kms key. This now allows the Cloudwatch Alarm to send notifications to the SNS topic.

### What Does This PR Do?

- Encode the KMS key (bfd-mgmt-cmk) and policy attached in Terraform by updating `ops/terraform/env/mgmt/kms-data-keys.tf` by adding a new 
resource `"aws_kms_key""bfd_mgmt_cmk"`
- Verified imported resources by running `terraform show`

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/env/mgmt/kms-data-keys.tf [hunk 1]
```diff
@@ -29,6 +29,30 @@ resource "aws_kms_alias" "data_keys" {
   target_key_id = aws_kms_key.data_keys[each.key].arn
 }
 
+resource "aws_kms_key" "bfd_mgmt_cmk" {
```
Comment (mjburling) [inline ops/terraform/env/mgmt/kms-data-keys.tf:32]: Can this key and its alias be rolled up adjacent to the existing data keys? e.g. 
```hcl
resource "aws_kms_key" "data_keys" {
  for_each = toset(concat(local.established_envs, ['mgmt']))
/*..snipped..*/
}

resource "aws_kms_alias" "data_keys" {
  for_each = toset(concat(local.established_envs, ["mgmt"]))
/*..snipped..*/
}
```

I haven't read this solution closely enough to make this a suggestion, so... it really is just a question–are there sufficient differences between the `mgmt` data cmk and the other data cmks to warrant separate definitions?

Comment (aadedejifearless) [reply ops/terraform/env/mgmt/kms-data-keys.tf:32]: I went with this suggestion and verified that the resources were imported.

File: ops/terraform/env/mgmt/kms-data-keys.tf
Comment (mjburling) [inline ops/terraform/env/mgmt/kms-data-keys.tf:1]: Cool. I think the tags definition for `aws_kms_key.data_keys` may need something like the following instead:
```hcl
  # Ensure mgmt definition doesn't conflict with identical default_tags with conditional
  tags = each.key != "mgmt" ? {
    Environment = each.key
    stack       = each.key
  } : {}
```

**Review by keithdadkins — APPROVED** [review]
Looks straightforward enough. Thanks!

**Review by mjburling — APPROVED** [review]

**Pull request merged**